reference_system_autoware/src/ros2/executor/autoware_default_singlethreaded.cpp

```diff
@@ -0,0 +1,168 @@
+                               .output_topic = "LanePlanner",
+                               .number_crunch_time = PROCESSING_TIME}));
+
+  // fusion nodes
+  constexpr auto FUSION_TIME = 100ms;
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "PointCloudFusion",
+                           .input_0 = "PointsTransformerFront",
+                           .input_1 = "PointsTransformerRear",
+                           .output_topic = "PointCloudFusion",
+                           .number_crunch_time = FUSION_TIME}));
+
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "NDTLocalizer",
+                           .input_0 = "VoxelGridDownsampler",
+                           .input_1 = "PointCloudMapLoader",
+                           .output_topic = "NDTLocalizer",
+                           .number_crunch_time = FUSION_TIME}));
+
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "NDTLocalizer",
+                           .input_0 = "VoxelGridDownsampler",
+                           .input_1 = "PointCloudMapLoader",
+                           .output_topic = "NDTLocalizer",
+                           .number_crunch_time = FUSION_TIME}));
+
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "VehicleInterface",
+                           .input_0 = "MPCController",
+                           .input_1 = "BehaviorPlanner",
+                           .output_topic = "VehicleInterface",
+                           .number_crunch_time = FUSION_TIME}));
+
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "Lanelet2GlobalPlanner",
+                           .input_0 = "rviz2",
+                           .input_1 = "NDTLocalizer",
+                           .output_topic = "Lanelet2GlobalPlanner",
+                           .number_crunch_time = FUSION_TIME}));
+
+  nodes.emplace_back(std::make_shared<node::Fusion>(
+      node::FusionSettings{.node_name = "Lanelet2MapLoader",
+                           .input_0 = "Lanelet2Map",
+                           .input_1 = "Lanelet2GlobalPlanner",
+                           .output_topic = "Lanelet2MapLoader",
+                           .number_crunch_time = PROCESSING_TIME}));
+
+  // reactor node
+  nodes.emplace_back(std::make_shared<node::Reactor>(node::ReactorSettings{
+      .node_name = "BehaviorPlanner",
+      .inputs = {"ObjectCollisionEstimator", "NDTLocalizer",
+                 "Lanelet2GlobalPlanner", "Lanelet2MapLoader", "ParkingPlanner",
+                 "LanePlanner"},
+      .output_topic = "BehaviorPlanner"}));
+
+  // command node
+  nodes.emplace_back(std::make_shared<node::Command>(node::CommandSettings{
+      .node_name = "VehicleDBWSystem", .input_topic = "VehicleInterface"}));
+
+  rclcpp::executors::SingleThreadedExecutor executor;
```

Comment:
Since the singlethreaded and multithreaded executor binary are just differing in one line I would propose that we move the whole content into a header called `reference_system.hpp` which consist of only one function:
```cpp
template<typename ExecutorType>
void create_and_start_reference_system(int argc, char *argv[]) {
  // code from main
  ExecutorType executor; // adjusted line 161
}
```

The main functions should then look like:
```cpp
int main() {
  create_and_start_reference_system<rclcpp::executors::SingleThreadExecutor>(argc, argv); // or multithread executor
}
```

which safes a lot of code duplication which can become a pain when we reconfigure the graph.

Reply:
great idea! ill add it to this PR